src/coreclr/jit/lowerxarch.cpp

```diff
@@ -1662,6 +1662,76 @@ GenTree* Lowering::LowerHWIntrinsic(GenTreeHWIntrinsic* node)
             LowerFusedMultiplyAdd(node);
             break;
 
+        case NI_SSE_And:
+        case NI_SSE2_And:
+        case NI_AVX_And:
+        case NI_AVX2_And:
+        case NI_AVX512F_And:
+        case NI_AVX512DQ_And:
+        case NI_SSE_Or:
+        case NI_SSE2_Or:
+        case NI_AVX_Or:
+        case NI_AVX2_Or:
+        case NI_AVX512F_Or:
+        case NI_AVX512DQ_Or:
+        case NI_SSE_Xor:
+        case NI_SSE2_Xor:
+        case NI_AVX_Xor:
+        case NI_AVX2_Xor:
+        case NI_AVX512F_Xor:
+        case NI_AVX512DQ_Xor:
```

Comment:
Can we introduce a sort of helper `gtIsBitwiseIntrinsic(intrinsic, oper)` so then a lot of paths in your PR can be simplified to `gtIsBitwiseIntrinsic(ni, GT_OR)`, `gtIsBitwiseIntrinsic(ni, GT_XOR)`, etc. ?

Reply:
Hi @EgorBo, thanks for the feedback. I am not sure I fully understand the methodology here, how could we apply `gtIsBitwiseIntrinsic(intrinsic, oper)` to the implementation? Do you mean we could extra the whole out of the switch statement, or we could possibly apply the helper to somewhere like `GetTernaryControlByte` to simplify the logics?

Reply:
Thanks for the explanation!